core/src/main/java/nu/yona/server/subscriptions/service/BuddyService.java

```diff
@@ -217,9 +217,9 @@ public BuddyDto addBuddyToAcceptingUser(UserDto acceptingUser, BuddyConnectReque
 
 	private Buddy createBuddyEntity(BuddyConnectRequestMessage connectRequestMessageEntity)
 	{
-		Buddy buddy = Buddy.createInstance(connectRequestMessageEntity.getSenderUser().get().getId(),
-				connectRequestMessageEntity.getFirstName(), connectRequestMessageEntity.getLastName(),
-				connectRequestMessageEntity.getSenderNickname(),
+		Optional<User> senderUser = connectRequestMessageEntity.getSenderUser();
+		Buddy buddy = Buddy.createInstance(senderUser.get().getId(), connectRequestMessageEntity.determineFirstName(senderUser),
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "senderUser.isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://build.dev.yona.nu:9000/coding_rules#rule_key=squid%3AS3655)

Reply:
In this context, we always have the sender user, which is why we were already ignoring this SonarQube comment.